custom_components/waste_collection_schedule/waste_collection_schedule/source/wellington_govt_nz.py

```diff
@@ -0,0 +1,70 @@
+import datetime
+from html.parser import HTMLParser
+
+import requests
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+from waste_collection_schedule.service.ICS import ICS
+
+
+TITLE = "Wellington City Council"
+DESCRIPTION = "Source for Wellington City Council."
+URL = "https://wellington.govt.nz"
+TEST_CASES = {
+    "Chelsea St": {"streetName": "chelsea street"},  # Friday
+    "Campbell St (ID Only)": {"streetId": "6515"},  # Wednesday
+}
+
+class Source:
+    def __init__(self, streetId= None, streetName= None):
+        self._streetId = streetId
+        self._streetName = streetName
+        self._ics = ICS()
+
+    def get_icon(self, wasteType):
+        if wasteType == "Rubbish Collection":
+            return "mdi:trash-can"
+        if wasteType == "Glass crate":
+            return "mdi:glass-fragile"
+        if wasteType == "Wheelie bin or recycling bags":
+            return "mdi:recycle"
+        return None
+
+    def get_image(self, wasteType):
+        if wasteType == "Rubbish Collection":
+            return "https://wellington.govt.nz/assets/images/rubbish-recycling/rubbish-bag.png"
+        if wasteType == "Glass crate":
+            return "https://wellington.govt.nz/assets/images/rubbish-recycling/glass-crate.png"
+        if wasteType == "Wheelie bin or recycling bags":
+            return "https://wellington.govt.nz/assets/images/rubbish-recycling/wheelie-bin.png"
+        return None
+        
+
+    def fetch(self):
+        # get token
+        if self._streetName:  
+            import json
+            url = "https://wellington.govt.nz/layouts/wcc/GeneralLayout.aspx/GetRubbishCollectionStreets"
+            headers = requests.structures.CaseInsensitiveDict()
+            headers["Origin"] = "https://wellington.govt.nz/rubbish-recycling-and-waste/when-to-put-out-your-rubbish-and-recycling"
+            headers["Access-Control-Request-Method"] = "POST"
+            headers["Content-Type"] = "application/json"
+            data = {"partialStreetName":self._streetName}
+            json_string = json.dumps(data)
+            r = requests.post(url, headers=headers, data=json_string)
+            data = json.loads(r.text)
+            if len(data["d"]) == 0:
+                raise Exception(f"No result found for {self._streetName}") 
+            if len(data["d"]) > 1: 
+                raise Exception(f"More then one result returned for {self._streetName}, be more specific or use StreetId instead")
```

Comment:
Case for "streetId" doesn't match.

Reply:
Added check for streetId a bit further which doubles to check if either streetid is returned by the the search or if it is supplied by the variable.